spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -2872,6 +2887,28 @@ def run_pygments_highlighter(self):
         if isinstance(self.highlighter, sh.PygmentsSH):
             self.highlighter.make_charlist()
 
+    def left_line(self, selection_position):
+        """Remove whitespace if we left a line with trailing whitespace,
+        and that we where not in a comment or string."""

```

Comment:
Per PEP 257, the first line of a docstring should be a one-line summary, followed by a blank line and additional description if necessary; the trailing ``"""`` needs to be on its own line if the docstring is not a oneliner, and per our conventions, multi-line docstrings should have the summary line on its own line (not the same line as the opening ``"""``. Furthermore, the actual text is unclear and ungrammatical. Also, it *does* remove whitespace (as it should) in comments, though not strings.

Therefore, I suggest:

``"""Remove trailing whitespace on leaving a non-string line containing it."""``